src/receiver.ts

```diff
@@ -1,7 +1,41 @@
-import { Message } from "./message";
+import { listen } from "socket.io";
+import { IMessageListener } from "./interfaces/iMessageListener";
 
+/**
+ * Handles the receiving of messages.
+ */
 export class Receiver {
-    public onMessage(message: Message) {
+    private readonly messageListener: IMessageListener;
+    private readonly server: any;
+    private readonly receivedMessages: string[];
 
+    constructor(messageListener: IMessageListener, port: string) {
+        this.messageListener = messageListener;
+        this.server = listen(port);
+        this.receivedMessages = [];
+
+        this.handleMessage();
+    }
+
+    /**
+     * Handles the incoming messages.
+     */
+    private handleMessage(): void {
+        // event fired every time a new client connects:
+        this.server.on("connection", (socket: any) => {
```

Comment:
Replace any by Socket from the socket.io lib

Reply:
This is not possible, the Server that socket.io exports is not the datatype used in that function.

Tried fixing this 'any' already and have not been successful.